x-pack/plugins/canvas/canvas_plugin_src/renderers/shape/index.js

```diff
@@ -22,7 +22,8 @@ export const shape = () => ({
 
     if (fill) shapeContent.setAttribute('fill', fill);
     if (border) shapeContent.setAttribute('stroke', border);
-    if (borderWidth >= 0) shapeContent.setAttribute('stroke-width', borderWidth);
+    const strokeWidth = Math.max(borderWidth, 0);
+    shapeContent.setAttribute('stroke-width', strokeWidth);
```

Comment:
This isn't relevant to the NaN viewBox issue, but I noticed when borderWidth < 0, there would be a border. This will bound the value to 0.